src/expression.rs

```diff
@@ -151,6 +154,33 @@ impl PartialEq for Expression {
 
 impl Eq for Expression {}
 
+macro_rules! impl_expr_op {
+    ($name:ident, $name_assign:ident, $function:ident, $function_assign:ident, $operator:ident) => {
+        impl $name for Expression {
+            type Output = Self;
+            fn $function(self, other: Self) -> Self {
+                Expression::Infix {
+                    left: Box::new(self),
+                    operator: InfixOperator::$operator,
+                    right: Box::new(other),
+                }
+            }
+        }
+        impl $name_assign for Expression {
+            fn $function_assign(&mut self, other: Self) {
+                let result = self.clone().$function(other);
```

Comment:
Is there some clever `std::mem` maneuver to sidestep the need for `clone()` here?